src/lib_io.c

```diff
@@ -166,6 +166,15 @@ static void io_file_readall(lua_State *L, FILE *fp)
     char *buf = lj_buf_tmp(L, m);
     n += (MSize)fread(buf+n, 1, m-n, fp);
     if (n != m) {
+      if (ferror(fp) != 0 && errno == EINTR) {
+        do {
+          n += (MSize)fread(buf+n, 1, m-n, fp);
```

Comment:
I think we should check `n == m` inside this `do ... while` loop as well?

Reply:
@agentzh 
I think if `ferror(fp) != 0` is `ture`, `n` should be smaller than `m`. So there is no need to add another check. If I am wrong, please let me know.